.github/workflows/terraform-stage.yml

```diff
@@ -0,0 +1,79 @@
+name: Run Terraform in staging
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths: [ 'terraform/**' ]
+
+defaults:
+  run:
+    working-directory: terraform/stage
+
+jobs:
+  terraform:
+    name: Terraform
+    runs-on: ubuntu-latest
+    environment: stage
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.TERRAFORM_STATE_ACCESS_KEY }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.TERRAFORM_STATE_SECRET_ACCESS_KEY }}
+      TF_VAR_cf_user: ${{ secrets.CF_USERNAME }}
+      TF_VAR_cf_password: ${{ secrets.CF_PASSWORD }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Terraform format
+        id: format
+        run: terraform fmt -check
+
+      - name: Terraform init
+        id: init
+        run: terraform init
+
+      - name: Terraform validate
+        id: validation
+        run: terraform validate -no-color
+
+      - name: Terraform plan
+        id: plan
+        run: terraform plan -no-color -input=false 2>&1 | tee plan_output.txt
+
+      - name: Read Terraform plan output file
+        id: terraform_output
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./terraform/stage/plan_output.txt
+
+      # inspiration: https://learn.hashicorp.com/tutorials/terraform/github-actions#review-actions-workflow
+      - name: Update PR
+        uses: actions/github-script@v4
+        # we would like to update the PR even when a prior step failed
+        if: ${{ always() }}
+        with:
+          script: |
+            const output = `Terraform Format and Style: ${{ steps.format.outcome }}
+            Terraform Initialization: ${{ steps.init.outcome }}
+            Terraform Validation: ${{ steps.validation.outcome }}
+            Terraform Plan: ${{ steps.plan.outcome }}
+
+            <details><summary>Show Plan</summary>
+
+            \`\`\`\n
+            ${{ steps.terraform_output.outputs.content }}
+            \`\`\`
+
+            </details>
+
+            *Pusher: @${{ github.actor }}, Action: \`${{ github.event_name }}\`*`;
+
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: output
+            })
+
+      - name: Terraform plan status
+        if: ${{ always() }} && steps.plan.outcome == 'failure'
```

Comment:
[question] This is currently sending an exit code 1, and I kinda wonder if it's because `always()` doesn't play well with && ? This is currently the way it is because I'm not using `continue-on-error` during the `terraform plan` step, although I think I would prefer that. I took it out after the comment someone made that `continue-on-error` seemed like code smell, but I think adding `always()` to the following actions is not my favorite, either.
Thoughts?

Reply:
I think the closing brackets around `always()` are tripping it up. If left in, this should either be

`if: ${{ always() && steps.plan.outcome == 'failure' }}` or `if: always() && steps.plan.outcome == 'failure'` or maybe even just `if: steps.plan.outcome == 'failure'`

[suggestion] this whole step can be removed. The job will fail properly when plan fails since you removed `continue-on-error`

Reply:
Oh yeah, I bet you're right :)  Thanks!